pytorch_lightning/plugins/precision/precision_plugin.py

```diff
@@ -75,6 +76,15 @@ def backward(
 
         return closure_loss
 
+    def pre_optimizer_step(
+        self, pl_module: LightningModule, optimizer: Optimizer, optimizer_idx: int, closure: Callable, **kwargs
+    ) -> bool:
+        """Hook to do something before each optimizer step."""
+        return True
```

Comment:
What's the meaning of the bool return value?

Reply:
It is used to bypass run_optimizer_step  as AMP is going to run self.scaler.step(optimizer).

It was the only way I found to make accelerator.optimizer_step plugin agnostic.